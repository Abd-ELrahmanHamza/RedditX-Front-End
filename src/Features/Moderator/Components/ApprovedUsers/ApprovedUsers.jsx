/* eslint-disable jsx-a11y/img-redundant-alt */
import { useState, useEffect } from "react";

import ModeratorRow from "../ModeratorRow/ModeratorRow";

import { BiSearch } from "react-icons/bi";

import {
  Container,
  SearchContainer,
  SearchInput,
  SearchIcon,
} from "./ApprovedUsers.styled";

/**
 * ApprovedUsers component that is used in User management
 * @returns {React.Component}  ApprovedUsers component that is used in User management
 */

const ApprovedUsers = ({ Moderator, communityName }) => {
  const [searchField, setSearchField] = useState("");
  const [filteredMonsters, setFilterMonsters] = useState(Moderator);

  useEffect(() => {
    const newFilteredMonsters = Moderator.filter((mod) => {
      return mod._id?.substring(3).toLowerCase().includes(searchField);
    });
    setFilterMonsters(newFilteredMonsters);
  }, [Moderator, searchField]);

  const onSearchChange = (event) => {
    const searchFieldString = event.target.value.toLowerCase();
    setSearchField(searchFieldString);
  };

  return (
    <>
      <SearchContainer>
        <SearchInput
          onChange={onSearchChange}
          placeholder="Search for a user"
        ></SearchInput>
        <SearchIcon>
          <BiSearch></BiSearch>
        </SearchIcon>
      </SearchContainer>
      <Container>
        {filteredMonsters.map((mod) => (
          <ModeratorRow
            communityName={communityName}
            key={mod.userName}
            Moderator={mod}
            approved={true}
          ></ModeratorRow>
        ))}
      </Container>
    </>
  );
};

export default ApprovedUsers;
