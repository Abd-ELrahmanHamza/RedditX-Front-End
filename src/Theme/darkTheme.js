/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
  },
  // The colors of the backgrounds in the website
  background: {
    primary: "#1a1a1b",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
  },
};

export default darkTheme;
