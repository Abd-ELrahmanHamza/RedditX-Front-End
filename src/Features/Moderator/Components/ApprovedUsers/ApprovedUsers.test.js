import { shallow, mount } from "enzyme";
import { enzymeFind } from "styled-components/test-utils";
import ApprovedUsers from "./ApprovedUsers";
import { ThemeProvider } from "styled-components";
import darkTheme from "Theme/darkTheme";
import { BrowserRouter } from "react-router-dom";
import { SearchInput } from "./ApprovedUsers.styled";

const Moderator = [
  {
    _id: "mohamedRomee",
  },
];
const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

const communityName = "t5_imagepro";

describe("ApprovedUsers Component", () => {
  it("this is a test for ApprovedUsers Component", () => {
    const wrapper = mount(
      <BrowserRouter>
        <ThemeProvider theme={darkTheme}>
          <ApprovedUsers Moderator={Moderator} communityName={communityName} />{" "}
        </ThemeProvider>
      </BrowserRouter>
    );

    const renderedComponent = enzymeFind(wrapper, SearchInput);
    expect(renderedComponent.prop("className")).toBeDefined();
  });
});
