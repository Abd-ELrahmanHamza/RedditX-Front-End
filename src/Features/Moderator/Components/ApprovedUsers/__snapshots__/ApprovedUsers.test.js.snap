// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApprovedUsers Component this is a test for ApprovedUsers Component 1`] = `
<Fragment>
  <styled.div>
    <styled.input
      onChange={[Function]}
      placeholder="Search for a user"
    />
    <styled.div>
      <BiSearch />
    </styled.div>
  </styled.div>
  <styled.div>
    <ModeratorRow
      Moderator={
        Object {
          "_id": "mohamedRomee",
        }
      }
      approved={true}
      communityName="t5_imagepro"
    />
  </styled.div>
</Fragment>
`;
